appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/jmx/MBeanMetadata.java

```diff
@@ -243,13 +240,13 @@ public MetricType getTypeRaw() {
     }
 
     public void setTags(List<XmlTag> tags) {
+        if (tags == null) {
+            throw new IllegalArgumentException("tags must not be null");
+        }
         this.tags = tags;
     }
 
     public void addTags(List<XmlTag> tags) {
-        if (this.tags == null) {
-            this.tags = new ArrayList<>();
-        }
         this.tags.addAll(tags);
```

Comment:
Here a null check for the parameter would be benefitial. Same reason as above.

Reply:
Field tags cannot be null. This field is set in constructor and even setter cannot set it to null.

Reply:
You are right.